src/Sidenav/TreeviewItem.tsx

```diff
@@ -0,0 +1,288 @@
+
+  /** Whether the submenu is expanded, used in Sidenav. */
+  expanded?: boolean;
+
+  /** Select the callback function for the current option  */
+  onSelect?: (eventKey: T, event: React.SyntheticEvent<HTMLElement>) => void;
+
+  title?: React.ReactNode;
+}
+
+const defaultProps: Partial<TreeviewItemProps> = {
+  as: 'li',
+  classPrefix: 'dropdown-item'
+};
+
+/**
+ * Tree View Node
+ * @see https://www.w3.org/TR/wai-aria-practices-1.2/#TreeView
+ */
+const TreeviewItem: RsRefForwardingComponent<'li', TreeviewItemProps> = React.forwardRef<
+  HTMLLIElement,
+  TreeviewItemProps
+>((props, ref) => {
+  const {
+    as: Component,
+    children,
+    divider,
+    panel,
+    disabled,
+    className,
+    style,
+    classPrefix,
+    tabIndex,
+    icon,
+    title,
+    eventKey,
+    onClick,
+    onSelect,
+    ...rest
+  } = props;
+
+  const parentTreeitem = useContext(TreeviewItemContext);
+  const treeControl = useContext(TreeControlContext);
+
+  const { rtl } = useCustom('DropdownMenu');
+
+  const { merge, withClassPrefix, prefix } = useClassNames(classPrefix);
+  const { openKeys = [], onOpenChange, onSelect: onSidenavSelect } = useContext(SidenavContext);
+
+  // WAI-ARIA treeitem
+  const isRootNode = !parentTreeitem;
+  const isParentNode = !!children;
+
+  const treeitemRef = useEnsuredRef<HTMLLIElement>(ref);
+  const treeitemId = useUniqueId('treeitem-');
+  const treeitemExpanded =
+    treeControl.expandedNodeIds.includes(rest.id ?? treeitemId) || openKeys.includes(eventKey);
+  const treeitemLevel = isRootNode ? 1 : parentTreeitem.level + 1;
+
+  const treeitemAriaAttributes: React.HTMLAttributes<HTMLLIElement> = {
+    role: 'treeitem',
+    'aria-level': treeitemLevel,
+    'aria-disabled': disabled
+  };
+
+  if (isParentNode) {
+    treeitemAriaAttributes['aria-expanded'] = treeitemExpanded;
+  }
+
+  // Whether this treeitem is selected
+  const active = false;
+
+  const classes = merge(
+    className,
+    withClassPrefix({
+      [`pull-${rtl ? 'left' : 'right'}`]: isParentNode,
+      [treeitemExpanded ? 'expand' : 'collapse']: isParentNode,
+      'with-icon': icon,
+      open: treeitemExpanded,
+      submenu: isParentNode,
+      active,
+      disabled,
+      // Whether this treeitem has focus
+      focus: treeControl.activeDescendantId === treeitemId
+    })
+  );
+
+  const activate = useCallback(
+    (event?: React.SyntheticEvent<HTMLLIElement>) => {
+      onSelect?.(eventKey, event);
+
+      if (isParentNode) {
+        onOpenChange?.(eventKey, event);
+      }
+    },
+    [eventKey, onSelect, isParentNode, onOpenChange]
+  );
+
+  const handleClick = useCallback(
+    (event: React.MouseEvent<HTMLLIElement>) => {
+      event.preventDefault();
+      event.stopPropagation();
+
+      if (disabled) {
+        return;
+      }
+
+      activate(event);
+      onSidenavSelect?.(eventKey, event);
+    },
+    [disabled, treeitemExpanded, isParentNode, activate, onSidenavSelect, eventKey]
```

Comment:
React Hook useCallback has unnecessary dependencies: 'isParentNode' and 'treeitemExpanded'. Either exclude them or remove the dependency array